### DISCUSSION TITLE
Use GPU from Mac M1 with SpaCy

### DISCUSSION THREAD

**Discussion by alvaromarlo** [explosion/spaCy#8404] [Help: Installation] [answered]
Hi, I followed the instructions from [https://spacy.io/usage#gpu](url) to use the M1 GPU from my mac with SpaCy and it doesn't works. The error message is always:

`ValueError: GPU is not accessible. Was the library installed correctly?`

Also I tried with other instructions from [https://adrianeboyd.github.io/installing-numpy-spacy-apple-m1/](url) and they didn't solve the problem, the error is the same. Could someone help me? Do @adrianeboyd knows why it doesn't work? There is any alternative to use the GPU with SpaCy?

I know that since 2019 NVIDIA does not cooperate with Apple and I suppose that because of that, it is not possible to use CUDA. That is the principal problem? Also, there is any alternative of CuPy for Mac?

**Comment by adrianeboyd** [comment, marked as answer]
Sorry, this GPU is not currently supported. GPU support in spacy currently requires cupy support, which is primarily for NVIDIA GPUs.

There may be some improved CPU options in the future, including custom ops within thinc or using Accelerate with the upcoming version of numpy.
